files/setup-011-fluentbit.sh

```diff
@@ -0,0 +1,246 @@
+        Format               json
+        Time_Key             time
+        Time_Format          %Y-%m-%dT%H:%M:%S.%L
+        Time_Keep            On
+
+    [PARSER]
+        Name                 cri
+        Format               regex
+        Regex                ^(?<time>[^ ]+) (?<stream>stdout|stderr) (?<logtag>[^ ]*) (?<message>.*)$
+        Time_Key             time
+        Time_Format          %Y-%m-%dT%H:%M:%S.%L%z
+
+    [PARSER]
+        Name                 syslog
+        Format               regex
+        Regex                ^\<(?<pri>[0-9]{1,5})\>1 (?<time>[^ ]+) (?<host>[^ ]+) (?<ident>[^ ]+) (?<pid>[-0-9]+) (?<msgid>[^ ]+) (?<extradata>(\[(.*)\]|-)) (?<message>.+)$
+        Time_Key             time
+        Time_Format          %Y-%m-%dT%H:%M:%S.%L
+        Time_Keep            On
+__FLUENTBIT_CONFIGMAP__
+
+cat > /root/config/fluentbit-ds.yaml << EOF
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: fluent-bit
+  namespace: vmware-system
+  labels:
+    k8s-app: fluent-bit
+    version: v1
+    kubernetes.io/cluster-service: "true"
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: fluent-bit
+        version: v1
+        kubernetes.io/cluster-service: "true"
+    spec:
+      containers:
+      - name: fluent-bit
+        image: ${FLUENTBIT_IMAGE}:${FLUENTBIT_VERSION}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 2020
+        resources:
+          requests:
+            cpu: 5m
+            memory: 10Mi
+          limits:
+            cpu: 50m
+            memory: 60Mi
+        volumeMounts:
+        - name: varlog
+          mountPath: /var/log
+        - name: varlogcontainers
+          mountPath: /var/log/containers
+        - name: varlibdockercontainers
+          mountPath: /var/lib/docker/containers
+          readOnly: true
+        - name: fluent-bit-config
+          mountPath: /fluent-bit/etc/
+      volumes:
+      - name: varlog
+        hostPath:
+          path: /var/log
+      - name: varlogcontainers
+        hostPath:
+          path: /var/log/containers
+      - name: varlibdockercontainers
+        hostPath:
+          path: /var/lib/docker/containers
+      - name: fluent-bit-config
+        configMap:
+          name: fluent-bit-config
+      serviceAccountName: fluent-bit
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+      - operator: Exists
+        effect: NoExecute
+      - operator: Exists
+        effect: NoSchedule
+  selector:
+    matchLabels:
+      k8s-app: fluent-bit
+      version: v1
+      kubernetes.io/cluster-service: "true"
+EOF
+
+kubectl apply -f /root/config/fluentbit-preperations.yaml
+kubectl apply -f /root/config/fluentbit-configmap.yaml
+kubectl apply -f /root/config/fluentbit-ds.yaml
+kubectl wait deployment --all --timeout=-1s --for=condition=Available -n vmware-system
```

Comment:
that's an infinite wait?

Reply:
That correct. Since it's the last script in the order of execution, it would wait for all objects in `ns` `vmware-system` to become available. I will remove it. Thx

Reply:
I will use `kubectl wait --for=condition=ready pod -l k8s-app=fluent-bit --timeout=-10s -n vmware-system` instead of.